library/tinkerpop/src/test/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopFederatedIT.java

```diff
@@ -0,0 +1,284 @@
+import org.apache.tinkerpop.gremlin.structure.Edge;
+import org.apache.tinkerpop.gremlin.structure.T;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import uk.gov.gchq.gaffer.cache.CacheServiceLoader;
+import uk.gov.gchq.gaffer.graph.Graph;
+import uk.gov.gchq.gaffer.tinkerpop.util.GafferPopFederatedTestUtil;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.atIndex;
+import static uk.gov.gchq.gaffer.tinkerpop.util.GafferPopTestUtil.TEST_CONFIGURATION_1;
+
+public class GafferPopFederatedIT {
+    private static Graph federatedGraph;
+    private static GafferPopGraph gafferPopGraph;
+
+    public static final String VERTEX_PERSON_1 = "p1";
+    public static final String VERTEX_PERSON_2 = "p2";
+    public static final String VERTEX_SOFTWARE_1 = "s1";
+    public static final String VERTEX_SOFTWARE_2 = "s2";
+    public static final String SOFTWARE_GROUP = "software";
+    public static final String PERSON_GROUP = "person";
+    public static final String CREATED_EDGE_GROUP = "created";
+    public static final String NAME_PROPERTY = "name";
+    public static final String WEIGHT_PROPERTY = "weight";
+
+    @Before
+    public void setUp() throws Exception {
+        federatedGraph = GafferPopFederatedTestUtil.setUpFederatedGraph(GafferPopFederatedIT.class);
+        gafferPopGraph = GafferPopGraph.open(TEST_CONFIGURATION_1, federatedGraph);
+    }
+
+    @After
+    public void tearDown() {
+        CacheServiceLoader.shutdown();
+    }
+
+    @Test
+    public void shouldConstructFederatedGafferPopGraph() {
+        // When
+        final Map<String, Object> variables = gafferPopGraph.variables().asMap();
+
+        // Then
+        assertThat(variables.get(GafferPopGraphVariables.SCHEMA))
+                .isSameAs(federatedGraph.getSchema());
+
+        assertThat(variables.get(GafferPopGraphVariables.USER))
+                .hasFieldOrPropertyWithValue("userId", "user01");
+    }
+
+    @Test
+    public void shouldGetVerticesById() {
+        // Given
+        GraphTraversalSource g = gafferPopGraph.traversal();
+
+        // When
+        List<Map<Object, Object>> vertex1 = g.V(VERTEX_PERSON_1).elementMap().toList();
+        List<Map<Object, Object>> vertex2 = g.V(VERTEX_SOFTWARE_2).elementMap().toList();
+
+        // expected
+        Map<Object, String> expectedVertex1 = new HashMap<>();
+        expectedVertex1.put(T.id, VERTEX_PERSON_1);
+        expectedVertex1.put(T.label, PERSON_GROUP);
+        expectedVertex1.put("name", "person1Name");
+
+        Map<Object, String> expectedVertex2 = new HashMap<>();
+        expectedVertex1.put(T.id, VERTEX_SOFTWARE_2);
+        expectedVertex1.put(T.label, SOFTWARE_GROUP);
+        expectedVertex1.put("name", "software2Name");
+
+        // Then
+        assertThat(vertex1)
+                .hasSize(1)
+                .first()
```

Comment:
Would a `.containsOnly(expectedVertex)` assert maybe be better for these types of checks?